crates/chimney-core/src/tls/manual.rs

```diff
@@ -0,0 +1,307 @@
+/// path exists and is accessible.
+fn validate_cert_path(path: &Path, file_type: &str) -> Result<PathBuf, ServerError> {
+    // Canonicalize the path (resolves symlinks, relative paths, and makes absolute)
+    // This also validates that the path exists and is accessible
+    let canonical = path.canonicalize().map_err(|e| {
+        ServerError::InvalidCertificateFile {
+            path: path.display().to_string(),
+            message: format!("Cannot access {file_type}: {e}"),
+        }
+    })?;
+
+    // Verify it's a file, not a directory
+    if !canonical.is_file() {
+        return Err(ServerError::InvalidCertificateFile {
+            path: path.display().to_string(),
+            message: format!("{file_type} path is not a file"),
+        });
+    }
+
+    // NOTE: We don't restrict to specific directories because users may have
+    // certificates in various locations (/etc/letsencrypt, ~/certs, etc.)
+    // The canonicalize() check ensures the file exists, is accessible, and
+    // all path components are safely resolved. Additional directory restrictions
+    // could be added in the future if needed.
+
+    Ok(canonical)
+}
+
+/// Load certificate chain from a PEM file
+pub fn load_certificate_chain(path: &Path) -> Result<Vec<CertificateDer<'static>>, ServerError> {
+    // Validate the path before opening
+    let safe_path = validate_cert_path(path, "certificate")?;
+
+    let file = File::open(&safe_path).map_err(|e| ServerError::InvalidCertificateFile {
+        path: path.display().to_string(),
+        message: e.to_string(),
+    })?;
+
+    let mut reader = BufReader::new(file);
+    let certs_result = certs(&mut reader).collect::<Result<Vec<_>, _>>();
+
+    certs_result.map_err(|e| ServerError::InvalidCertificateFile {
+        path: path.display().to_string(),
+        message: format!("Failed to parse certificate: {e}"),
+    })
+}
+
+/// Load private key from a PEM file (supports RSA and ECDSA)
+pub fn load_private_key(path: &Path) -> Result<PrivateKeyDer<'static>, ServerError> {
+    // Validate the path before opening
+    let safe_path = validate_cert_path(path, "private key")?;
+
+    let file = File::open(&safe_path).map_err(|e| ServerError::InvalidPrivateKeyFile {
+        path: path.display().to_string(),
+        message: e.to_string(),
+    })?;
+
+    let mut reader = BufReader::new(file);
+    let key = private_key(&mut reader)
+        .map_err(|e| ServerError::InvalidPrivateKeyFile {
+            path: path.display().to_string(),
+            message: format!("Failed to parse private key: {e}"),
+        })?
+        .ok_or_else(|| ServerError::InvalidPrivateKeyFile {
+            path: path.display().to_string(),
+            message: "No private key found in file".to_string(),
+        })?;
+
+    return Ok(key);
+}
+
+/// Load a certified key from certificate and key files
+pub fn load_certified_key(
+    cert_file: &Path,
+    key_file: &Path,
```

Comment:
The `_ca_file` parameter is unused. If CA bundles aren't supported yet, consider:

1. Removing the parameter entirely until needed
2. Or returning an error if a CA file is provided:

```rust
pub fn load_certified_key(
    cert_file: &Path,
    key_file: &Path,
    ca_file: Option<&Path>,
) -> Result<Arc<CertifiedKey>, ServerError> {
    if let Some(ca) = ca_file {
        return Err(ServerError::TlsInitializationFailed(
            format!("CA bundles not yet supported: {}", ca.display())
        ));
    }
    // ... rest of implementation
}
```

This makes it explicit that the feature isn't implemented rather than silently ignoring it.